### PR TITLE
Add Customers CRUD

### PR DESCRIPTION
**Pull request by EduardoCitelli**
# Feature

Customers CRUD

# Description

Create Endpoints and controller for Customers entities

# Changelog

- add customer entity
- add customer controller
- add customer routing

### CODE REVIEW

File: services/caldar-api/src/controllers/customer-controller.ts [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+  "data",
+  "customers.json"
+);
+
+type queryType = {
+  type?: CustomerType;
+};
+
+export const getCustomers: RequestHandler = (req: Request, res: Response) => {
+  let response = getDataCustomers();
```
Comment (juanzitelli) [inline services/caldar-api/src/controllers/customer-controller.ts:19]: Let's try to use `const` when a variable is not re-assigned

```suggestion
  const response = getDataCustomers();
```

File: services/caldar-api/src/controllers/customer-controller.ts [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+export const getCustomersByType: RequestHandler = (
+  req: Request,
+  res: Response
+) => {
+  const query: queryType = {};
+
+  query.type = req.query.type ? Number(req.query.type) : undefined;
+
+  let customers = getDataCustomers();
```
Comment (juanzitelli) [inline services/caldar-api/src/controllers/customer-controller.ts:35]: Let's try to use `const` when a variable is not re-assigned
```suggestion
  const customers = getDataCustomers();
```

File: services/caldar-api/src/controllers/customer-controller.ts [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+  saveData(updatedList);
+
+  res.status(200).json({
+    status: "ok",
+    content: deletedCustomer,
+  });
+};
+
+function saveData(list: Customer[]) {
```
Comment (juanzitelli) [inline services/caldar-api/src/controllers/customer-controller.ts:145]: Let's try to use arrow functions when possible! To keep consistency
```suggestion
const saveData = (list: Customer[]) => {
```

File: services/caldar-api/src/controllers/customer-controller.ts [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+    customersDataFilePath,
+    JSON.stringify(list, null, 2),
+    "utf8"
+  );
+}
+
+function getDataCustomers(): Customer[] {
+  let data = fs.readFileSync(customersDataFilePath, "utf-8");
+
+  let response: Customer[] = JSON.parse(data);
```
Comment (juanzitelli) [inline services/caldar-api/src/controllers/customer-controller.ts:156]:
```suggestion
  const data = fs.readFileSync(customersDataFilePath, "utf-8");

  const response: Customer[] = JSON.parse(data);
```

File: services/caldar-api/src/controllers/customer-controller.ts [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+  req: Request,
+  res: Response
+) => {
+  const query: queryType = {};
+
+  query.type = req.query.type ? Number(req.query.type) : undefined;
+
+  let customers = getDataCustomers();
+
+  let list = !query.type
```
Comment (juanzitelli) [inline services/caldar-api/src/controllers/customer-controller.ts:37]: Let's try to use `const` when a variable is not re-assigned
```suggestion
  const list = !query.type
```

File: services/caldar-api/src/controllers/customer-controller.ts [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+    res.status(400).json({
+      status: "error",
+      error:
+        "The ID of the customer to update must be present in the request body.",
+    });
+  }
+
+  let customers = getDataCustomers();
+
+  let existCustomer = customers.some((x) => x.id === updatedCustomer.id);
```
Comment (juanzitelli) [inline services/caldar-api/src/controllers/customer-controller.ts:91]: Replace `let` for `const` and rename `existsCustomer` boolean variable for a better spelling

```suggestion
  const customers = getDataCustomers();

  const customerExists = customers.some((x) => x.id === updatedCustomer.id);
```

File: services/caldar-api/src/controllers/customer-controller.ts [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+
+  if (!existCustomer) {
```
Comment (juanzitelli) [inline services/caldar-api/src/controllers/customer-controller.ts:93]: Renamed variable for a better spelling
```suggestion
  if (!customerExists) {
```

File: .gitignore [hunk 8]
```diff
@@ -115,4 +115,5 @@ dist
 .yarn/unplugged
 .yarn/build-state.yml
 .yarn/install-state.gz
-.pnp.*
\ No newline at end of file
+.pnp.*
+/.vs
```
Comment (juanzitelli) [inline .gitignore:119]: If with this line you were trying to ignore .vscode, please don't. I intentionally added a `.vscode` extension recommendation file so the IDE would suggest you install a REST client which I think it's nice!.
```suggestion
```

File: packages/caldar-types/src/enums/customer-type.ts [hunk 9]
```diff
@@ -0,0 +1,4 @@
+export enum CustomerType {
+  CONSTRUCTION = 1,
+  PRIVATE,
```
Comment (juanzitelli) [inline packages/caldar-types/src/enums/customer-type.ts:3]: When declare other `enums` (i.e. `CustomerRoutes`) we use the "Pascal" casing convention, what do you think about changing this one for it to respect that convention?
```suggestion
  Construction = 1,
  Private,
```

**Review by juanzitelli — CHANGES_REQUESTED** [review]
Nice job! 👏🏻 Still, please take a look at the comments I left below, you can commit them directly in Github without actually going back to VSCode!

File: .gitignore (hunk 8 above)
Comment (EduardoCitelli) [reply .gitignore:119]: this ignore si for de IDE visual studio 2019

**Review by juanzitelli — APPROVED** [review]
Nice job 👏🏻

**Pull request merged**